### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will clean up a number of warnings from various compilers and situations
and update the toml11 library.

### CODE REVIEW

File: src/helics/application_api/HelicsPrimaryTypes.hpp [hunk 1]
```diff
@@ -34,14 +34,14 @@ using defV = std::variant<double,
                           NamedPoint>;
 
 /**enumeration of the order inside the variant so the Which function returns match the enumeration*/
```
Comment (nightlark) [inline src/helics/application_api/HelicsPrimaryTypes.hpp:36]: Unrelated to the warnings, but I'm a bit confused what this comment is saying.

Is it something like "enumeration of the order inside the variant so the order of the function returns match the enumeration"?

File: src/helics/apps/AsioBrokerServer.cpp [hunk 2]
```diff
@@ -94,7 +94,7 @@ namespace udp {
 
         asio::ip::udp::socket mSocket;
         asio::ip::udp::endpoint mRemoteEndpoint;
-        std::array<char, 1024> mRecvBuffer{0};
+        std::array<char, 1024> mRecvBuffer{{0}};
```
Comment (nightlark) [inline src/helics/apps/AsioBrokerServer.cpp:97]: Is this something like nested value initialization, with the inner `{0}` specifying the value for the individual char entries, and `{{0}}` setting the initial value for the entire array to 0's?

File: src/helics/apps/AsioBrokerServer.cpp [hunk 3]
```diff
@@ -270,10 +270,12 @@ namespace apps {
     void AsioBrokerServer::mainLoop()
     {
         auto ioctx = AsioContextManager::getContextPointer();
+
 #ifdef HELICS_ENABLE_TCP_CORE
         if (tcp_enabled_) {
             tcpserver = loadTCPserver(ioctx->getBaseContext());
             tcpserver->setDataCall(
+                // NOLINTNEXTLINE
```
Comment (nightlark) [inline src/helics/apps/AsioBrokerServer.cpp:278]: What warning was caused by the next line?

File: src/helics/application_api/HelicsPrimaryTypes.hpp (hunk 1 above)
Comment (phlptp) [reply src/helics/application_api/HelicsPrimaryTypes.hpp:36]: yes

File: src/helics/apps/AsioBrokerServer.cpp (hunk 3 above)
Comment (phlptp) [reply src/helics/apps/AsioBrokerServer.cpp:278]: clang-tidy warning about the parameter being able to be passed by reference but it is a shared pointer that needs its lifetime extended so it shouldn't be passed by reference.

File: src/helics/apps/AsioBrokerServer.cpp (hunk 2 above)
Comment (phlptp) [reply src/helics/apps/AsioBrokerServer.cpp:97]: one of the clang compilers gave a warning on this, something about partial initializer lists and recommended this fix.

Comment (phlptp) [reply src/helics/apps/AsioBrokerServer.cpp:97]: and no it is just the first value that is set to 0

**Review by nightlark — APPROVED** [review]

**Pull request merged**